website/docs/r/nsxt_edgegateway.html.markdown

```diff
@@ -0,0 +1,177 @@
+  name = "nsxt-edge"
+}
+
+resource "vcd_nsxt_edgegateway" "nsxt-edge" {
+  org                     = "my-org"
+  vdc                     = "nsxt-vdc"
+  name                    = "nsxt-edge"
+  description             = "Description"
+
+  external_network_id = data.vcd_external_network_v2.nsxt-ext-net.id
+
+  subnet {
+     gateway               = "10.150.191.253"
+     prefix_length         = "19"
+     # primary_ip should fall into defined "allocated_ips" range as otherwise
+     # next apply will report additional range of "allocated_ips" with the range
+     # containing single "primary_ip" and will cause non-empty plan.
+     primary_ip            = "10.150.160.137"
+     allocated_ips {
+       start_address = "10.150.160.137"
+       end_address   = "10.150.160.138"
+     }
+  }
+}
+```
+
+## Example Usage (Using custom Edge Cluster and multiple subnets)
+
+```hcl
+data "vcd_nsxt_edge_cluster" "secondary" {
+  name = "edge-cluster-two"
+}
+
+
+data "vcd_external_network_v2" "nsxt-ext-net" {
+  name = "nsxt-edge"
+}
+
+resource "vcd_nsxt_edgegateway" "nsxt-edge" {
+  org                     = "my-org"
+  vdc                     = "nsxt-vdc"
+  name                    = "nsxt-edge"
+  description             = "Description"
+
+  external_network_id = data.vcd_external_network_v2.nsxt-ext-net.id
+
+  # Custom edge cluster reference
+  edge_cluster_id = data.vcd_nsxt_edge_cluster.secondary.id
+
+  subnet {
+     gateway               = "10.150.191.253"
+     prefix_length         = "19"
+     # primary_ip should fall into defined "allocated_ips" range as otherwise
+     # next apply will report additional range of "allocated_ips" with the range
+     # containing single "primary_ip" and will cause non-empty plan.
+     primary_ip            = "10.150.160.137"
+     allocated_ips {
+       start_address = "10.150.160.137"
+       end_address   = "10.150.160.137"
+     }
+  }
+
+  subnet {
+     gateway       = "77.77.77.1"
+     prefix_length = "26"
+
+     allocated_ips {
+       start_address = "77.77.77.10"
+       end_address   = "77.77.77.12"
+     }
+  }
+
+  subnet {
+     gateway       = "88.88.88.1"
+     prefix_length = "24"
+
+     allocated_ips {
+       start_address = "88.88.88.91"
+       end_address   = "88.88.88.92"
+     }
+
+     allocated_ips {
+       start_address = "88.88.88.94"
+       end_address   = "88.88.88.95"
+     }
+
+     allocated_ips {
+       start_address = "88.88.88.97"
+       end_address   = "88.88.88.98"
+     }
+  }
+}
+```
+
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - (Optional) The name of organization to which the VDC belongs. Optional if defined at provider level.
+* `vdc` - (Optional) The name of VDC that owns the edge gateway. Optional if defined at provider level.
+* `name` - (Required) A unique name for the edge gateway.
+* `description` - (Optional) A unique name for the edge gateway.
+* `external_network_id` - (Required) A unique name for the edge gateway.
```

Comment:
should this be "An external network ID"?

Reply:
Thanks. Fixed.